pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -354,16 +356,33 @@ private boolean sendAutomaticSavepoint(Query query, int flags) throws IOExceptio
         // If CompositeQuery is observed, just assume it might fail and set the savepoint
         || !(query instanceof SimpleQuery)
         || ((SimpleQuery) query).getFields() != null)) {
+
       sendOneQuery(autoSaveQuery, SimpleQuery.NO_PARAMETERS, 1, 0,
           QUERY_NO_RESULTS | QUERY_NO_METADATA
               // PostgreSQL does not support bind, exec, simple, sync message flow,
               // so we force autosavepoint to use simple if the main query is using simple
               | QUERY_EXECUTE_AS_SIMPLE);
+
       return true;
     }
     return false;
   }
 
+  private void releaseSavePoint(boolean autosave, int flags) throws SQLException {
+    if (autosave
+        && (flags & QueryExecutor.QUERY_SUPPRESS_BEGIN) == 1
```

Comment:
After changing the check for the QUERY_SUPPRESS_BEGIN to 0 the changes fix the issue. Otherwise the savepoint  is never released

Reply:
Sadly that doesn't pass the test suite...